packages/engine/lib/storage/src/lib.rs

```diff
@@ -1,3 +1,27 @@
+//! Library for low-level to store and load arrow-data.
```

Comment:
Sorry it does not, I agree. It stores things on disk, which is a form of storage. That's why I'm wondering if this module name is perhaps misleading.

Apache Arrow describes itself as a memory format, so I'm not sure `memory` is misleading.

Reply:
Hmm, maybe we should split this logic into two crates then. One is called `memory`, one is called `memory_format`? I think this would avoid confusion.

Reply:
I don't think adding yet another crate is necessary here. I think we can just call the crate `memory`. Including `memory` format logic within it is fine, there are modules within the crate describing the separation. 

The only problem before was `storage` and file-system, but we're not going to include that stuff in this crate. At least not for now, maybe later on we do want to do that instead of `output`, but we can re-evaluate the naming at that point.

Reply:
When renaming `storage` -> `memory`, I'd rename the `memory` module to `shared_memory`. Do we want to rename `arrow` to `format`?

Reply:
Looked through the files, I think `shared_memory` would be an okay rename for the module. Lets not call `arrow` `format`, I think it's less clear

Reply:
Renamed it in e6c735601fd674e55d8678d6b3c0e810eab86bcd